runtime/src/lib.rs

```diff
@@ -173,12 +113,47 @@ fn handle_command(drivers: &mut Drivers) -> CaliptraResult<MboxStatusE> {
     }
 }
 
+fn handle_and_respond(drivers: &mut Drivers) -> CaliptraResult<MboxStatusE> {
+    // For firmware update, don't read data from the mailbox
+    if drivers.mbox.cmd() == CommandId::FIRMWARE_LOAD.into() {
+        update::handle_impactless_update(drivers)?;
+
+        // If the handler succeeds but does not invoke reset that is
+        // unexpected. Denote that the update failed.
+        return Err(CaliptraError::RUNTIME_UNEXPECTED_UPDATE_RETURN);
+    }
+
+    // Get the command bytes
+    let req_packet = Packet::copy_from_mbox(drivers)?;
+    let cmd_bytes = req_packet.as_bytes()?;
+
+    cprintln!(
+        "[rt] Received command=0x{:x}, len={}",
+        req_packet.cmd,
+        req_packet.len
+    );
+
+    // Create a response packet
+    // TODO: OPEN: How big is our stack? We are throwing 8k on it between these 2 packets
+    let mut resp_packet = Packet::default();
```

Comment:
Do u mean copying responses directly to mailbox data memory bypassing the register interface?

Reply:
No, I mean taking a response struct, calling as_bytes() and writing it to the mailbox instead of first copying it to a Packet structure.

Reply:
perhaps add some scratch buffers in the environment (Drivers) with static life time.

Reply:
That sounds like it could be a good long-term solution. I'm open to working on that but I may need a little hand-holding to do it right. For now, Jordan pointed me to a similar example using an enum of structs. It doesn't fully solve the stack allocation issue, but it at least limits it to the smallest size possible. I'll push a mostly complete example of that soon.